cmd/dockerd/daemon.go

```diff
@@ -861,11 +867,20 @@ func configureDaemonLogs(conf *config.Config) {
 	} else {
 		logrus.SetLevel(logrus.InfoLevel)
 	}
-	logrus.SetFormatter(&logrus.TextFormatter{
-		TimestampFormat: jsonmessage.RFC3339NanoFixed,
-		DisableColors:   conf.RawLogs,
-		FullTimestamp:   true,
-	})
+	var formatter logrus.Formatter
+	switch conf.LogFormat {
+	case log.JSONFormat:
+		formatter = &logrus.JSONFormatter{
+			TimestampFormat: jsonmessage.RFC3339NanoFixed,
+		}
+	default:
```

Comment:
Since we validate `JSONFormat` or `TextFormat`, I think we should make that explicit here instead of relying on `default:`.

We could just make the default case a panic instead.

Reply:
Ok - I don't typically use panic but happy to make the change.